src/perspective.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//     You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(internal, scope) {
+
+  function Perspective(length) {
+    if (arguments.length != 1) {
+      throw new TypeError('Perspective takes exactly 1 argument.');
+      // TODO: Change check of 'px' when LengthValue.LengthType enum is updated.
+    } else if (!(length instanceof SimpleLength) || length.type != 'px') {
```

Comment:
Since the later ifs can't be reached if one of the others is matched, these can be
```javascript
if (condition) {
  throw ...
}
if (condition) {
  throw
}
```

This is enforced by the style guide in C++ so may as well get used to it now :)